flink-cdc-e2e-tests/flink-cdc-pipeline-e2e-tests/src/test/java/org/apache/flink/cdc/pipeline/tests/MySqlToElasticsearchE2eITCase.java

```diff
@@ -0,0 +1,231 @@
+        LOG.info("Elasticsearch container is started.");
+    }
+
+    @After
+    public void after() {
+        super.after();
+        mysqlInventoryDatabase.dropDatabase();
+        dropElasticsearchIndex("customers");
+        dropElasticsearchIndex("products");
+        if (elasticsearch != null) {
+            elasticsearch.stop();
+        }
+    }
+
+    @Test
+    public void testSyncWholeDatabase() throws Exception {
+        String pipelineJob = String.format(
+                "source:\n"
+                        + "  type: mysql\n"
+                        + "  hostname: mysql\n"
+                        + "  port: 3306\n"
+                        + "  username: %s\n"
+                        + "  password: %s\n"
+                        + "  tables: %s.\\.*\n"
+                        + "  server-id: 5400-5404\n"
+                        + "  server-time-zone: UTC\n"
+                        + "\n"
+                        + "sink:\n"
+                        + "  type: elasticsearch\n"
+                        + "  hosts: http://elasticsearch:9200\n"
+                        + "  version: %s\n"
+                        + "  batch.size.max.bytes: 5242880\n"
+                        + "  record.size.max.bytes: 5242880\n"
+                        + "\n"
+                        + "pipeline:\n"
+                        + "  parallelism: 1",
+                MYSQL_TEST_USER,
+                MYSQL_TEST_PASSWORD,
+                mysqlInventoryDatabase.getDatabaseName(),
+                getElasticsearchVersion().split("\\.")[0]  // Use major version number
+        );
+
+        Path mysqlCdcJar = TestUtils.getResource("mysql-cdc-pipeline-connector.jar");
+        Path elasticsearchCdcConnector = TestUtils.getResource("flink-cdc-pipeline-connector-elasticsearch-3.2-SNAPSHOT.jar");
+        Path mysqlDriverJar = TestUtils.getResource("mysql-driver.jar");
+
+        submitPipelineJob(pipelineJob, mysqlCdcJar, elasticsearchCdcConnector, mysqlDriverJar);
+        waitUntilJobRunning(Duration.ofSeconds(30));
+        LOG.info("Pipeline job is running");
+
+        // Wait for data to be written
+        Thread.sleep(20000);
+        validateElasticsearchContent("customers");
+        validateElasticsearchContent("products");
+
+        LOG.info("Begin incremental reading stage.");
+        String mysqlJdbcUrl = String.format(
+                "jdbc:mysql://%s:%s/%s",
+                MYSQL.getHost(),
+                MYSQL.getDatabasePort(),
+                mysqlInventoryDatabase.getDatabaseName()
+        );
+
+        try (Connection conn = DriverManager.getConnection(mysqlJdbcUrl, MYSQL_TEST_USER, MYSQL_TEST_PASSWORD);
+             Statement stat = conn.createStatement()) {
+
+            stat.execute("INSERT INTO products VALUES (default,'jacket','water resistent white wind breaker',0.2, null, null, null);");
+
+            // Wait for data to be written
+            Thread.sleep(20000);
```

Comment:
Would prefer putting retry and timeout strategies in `validateElasticsearchContent` (just like `waitUntilSpecificEvent` in other E2e cases) instead of sleeping for a fixed duration, which will make tests finish faster if results are as expected.